vignettes/miaViz.Rmd

```diff
@@ -326,6 +326,127 @@ names(metadata(dmn_se))
 plotDMNFit(dmn_se, type = "laplace")
 ```
 
+# Serial data ordination and trajectories
+
+```{r, error=FALSE, warning=FALSE, results='hide'}
+if (!requireNamespace("devtools", quietly = TRUE))
+    install.packages("devtools")
+devtools::install_github("microbiome/miaTime")
+```
+
+Principal Coordinate Analysis using Bray-Curtis dissimilarity on the
+`hitchip1006` dataset:
+
+```{r}
+library(dplyr)
+data(hitchip1006, package = "miaTime")
+tse <- hitchip1006
+tse <- transformCounts(tse, method = "relabundance")
+## Ordination with PCoA with Bray-Curtis dissimilarity
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+                  name = "PCoA_BC", exprs_values = "relabundance",
+              na.rm = TRUE)
+# Calculating Explained variance
+e <- attr(reducedDim(tse, "PCoA_BC"), "eig")
+rel_eig <- e/sum(e[e>0])
+# plot
+p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
+    labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]],1), "%", ")", sep = ""),
+         y = paste("PCoA 2 (", round(100 * rel_eig[[2]],1), "%", ")", sep = ""),
+         title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+p1
+```
+
+Retrieving information about all available trajectories:
+
+```{r}
+colData(tse)$time <- as.factor(colData(tse)$time)
+# trajectories info
+traj_info <- table(colData(tse)$subject,colData(tse)$time)
+# has at least two points
+traj_info <- traj_info[rowSums(traj_info)!=1,]
+table(rowSums(traj_info)) %>% as.data.frame() %>%
+    `colnames<-`(c("Trajectories Lengths", "Counts"))
+```
+
+Lets look at all present trajectories in the data:
+
+```{r}
+# keeping only data that has one-to-one link from one point to another
+df <- traj_info %>% as.data.frame() %>%
+    `colnames<-`(c("subject", "time", "value")) %>% 
+    filter(value ==1) %>% arrange(subject, time) 
+# retrieving corresponding ordination coordinate values
+df <- cbind(df, 
+      reducedDim(
+          tse[,(colData(tse)$subject %in% df$subject &
+                    colData(tse)$time %in% df$time)], "PCoA_BC") %>%
+          `colnames<-`(c("PC1","PC2")) %>% as_tibble(rownames = "sample_id") %>% 
+          mutate(subject=colData(tse)[sample_id,"subject"]) %>%
+          arrange(subject)) %>% select(-last_col())
+# retrieving the corresponding start and end of each trajectory 
+df_start_end <- rbind(
+    df %>% group_by(subject) %>% slice(which.min(time)) %>% ungroup() %>%
+        mutate(tp=rep("start",nrow(.))),
+    df %>% group_by(subject) %>% slice(which.max(time)) %>% ungroup() %>%
+        mutate(tp=rep("end",nrow(.)))
+)
```

Comment:
I thought having the information about start and end of a trajectory, could be useful in a case where for instance points are colored with cluster ids after a clustering analysis; then could inform about movement from a cluster to another by the time? 
I will remove it in this case, but was just curious in general if this could be a good feature to add if to be implemented later as a function?

Reply:
alright, I see now that arrow at geom_path will do it, thanks haven't spotted it earlier.

Reply:
We may like to include indications on cluster switchings etc in analyses but that is a different task. In the current one let us exclusively focus on trajectories for pedagogical reasons.